src/savegame_json.cpp

```diff
@@ -1490,6 +1490,11 @@ void item::io( Archive& archive )
     contents.erase( std::remove_if( contents.begin(), contents.end(), []( const item &cont ) {
         return cont.is_null();
     } ), contents.end() );
+
+    // Sealed item migration: items with "unseals_into" set should always have contents
+    if( contents.empty() && is_non_resealable_container() ) {
+        on_contents_changed();
```

Comment:
Perform your migration here explicitly. A future PR could (and probably will) extend `item::on_contents_changed` and that could cause some excitement. Savegame migration code should be as concise as possible to avoid possible catastrophe.